src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2383 @@
+				(*iterModes).GenerateSolutionSpace(ResolutionMsa, ResolutionOSA);
+			}
+
+			Initialized = true;
+		}
+
+
+		Real64 Model::Part_press(Real64 P, Real64 W)
+		{
+			// Function to compute partial vapor pressure in [kPa]
+			// From page 6.9 equation 38 in ASHRAE Fundamentals handbook (2005)
+			//   P = ambient pressure [kPa]
+			//   W = humidity ratio [kg/kg dry air]
+
+			return (P * W / (0.62198 + W));
+		}
+		
+		Real64 Model::Sat_press(Real64 Tdb)
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Function to compute saturation vapor pressure in [kPa]
+
+			// METHODOLOGY EMPLOYED:
+			//ASHRAE Fundamentals handbood (2005) p 6.2, equation 5 and 6
+			//   Tdb = Dry bulb temperature [degC]
+			// Valid from -100C to 200 C
+
+			// REFERENCES:
+			// na
+
+			Real64  C1 = -5674.5359;
+			Real64  C2 = 6.3925247;
+			Real64  C3 = -0.009677843;
+			Real64  C4 = 0.00000062215701;
+			Real64  C5 = 2.0747825E-09;
+			Real64  C6 = -9.484024E-13;
+			Real64  C7 = 4.1635019;
+			Real64  C8 = -5800.2206;
+			Real64  C9 = 1.3914993;
+			Real64  C10 = -0.048640239;
+			Real64  C11 = 0.000041764768;
+			Real64  C12 = -0.000000014452093;
+			Real64  C13 = 6.5459673;
+			Real64  Sat_press_val = 0;
+
+			Real64   TK = Tdb + 273.15;         //Converts from degC to degK
+
+			if (TK <= 273.15)
+			{
+				Sat_press_val = exp(C1 / TK + C2 + C3 * TK + C4 * pow(TK, 2) + C5 * pow(TK, 3) + C6 * pow(TK, 4) + C7 * log(TK)) / 1000;
+			}
+			else
+			{
+				Sat_press_val = exp(C8 / TK + C9 + C10 * TK + C11 * pow(TK, 2) + C12 * pow(TK, 3) + C13 * log(TK)) / 1000;
+			}
+			return Sat_press_val;
+
+		}
+		
+
+		Real64 Model::CalcHum_ratio_W(Real64 Tdb, Real64 RH, Real64 P)
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Function to calculate humidity ratio [kg H2O/kg air]
+			// Given dry bulb and wet bulb temperature inputs [degC]
+
+			// METHODOLOGY EMPLOYED:
+			// ASHRAE Fundamentals handbood (2005)
+			// Tdb = Dry bulb temperature [degC]
+			// RH = Relative Humidity [Fraction or %]
+			// P = Ambient Pressure [kPa] 
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+
+			Real64 Pws = Sat_press(Tdb);
+			Real64 Hum_rat = 0.62198 * RH * Pws / (P - RH * Pws);   // Equation 22, 24, p6.8
+			return Hum_rat;
+		}
+
+		Real64 Model::CheckVal_W(Real64 W, Real64 T, Real64 P)
+		{
+			Real64 OutletRHtest = PsyRhFnTdbWPb(T, W, P); //could also use outlet pressure instead of fixed
+			Real64 OutletW = PsyWFnTdbRhPb(T, OutletRHtest, P); 
+			if (OutletRHtest == 1)
+			{
+				if (OutletW < Wsa)
+				{
+					ShowWarningError("Humidity ratio exceeded realistic range, check performance curve");

```

Comment:
Is this warning going to be thrown many times?  If so, it will make the error file unreadable, so you may need to trap for multiple calls to it.

Reply:
fixed